### PR TITLE
[Broker] Fix call sync method in async rest api for internalGetBacklogSizeByMessageId

### PR DESCRIPTION
**Pull request by liudezhi2098**
### Motivation

Avoid call sync method in async rest API for PersistentTopicsBase#internalGetBacklogSizeByMessageId.

### Modifications

- *Use async instead of sync method.*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Need to update docs? 
  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -2783,49 +2783,55 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
...
+        future.thenAccept(__ -> {
+            getPartitionedTopicMetadataAsync(topicName, authoritative, false)
+                    .thenAccept(partitionMetadata -> {
+                        if (!topicName.isPartitioned() && partitionMetadata.partitions > 0) {
+                            log.warn("[{}] Not supported calculate backlog size operation on partitioned-topic {}",
+                                    clientAppId(), topicName);
+                            asyncResponse.resume(new RestException(Status.METHOD_NOT_ALLOWED,
+                                    "calculate backlog size is not allowed for partitioned-topic"));
+                        } else {
+                            validateTopicOwnership(topicName, authoritative);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2801]: this is a sync method need to fix.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -2783,49 +2783,55 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
...
+                            validateTopicOperation(topicName, TopicOperation.GET_BACKLOG_SIZE);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2802]: same above.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 3]
```diff
@@ -2783,49 +2783,55 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
...
+                            PersistentTopic topic = (PersistentTopic) getTopicReference(topicName);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2803]: same above.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 4]
```diff
@@ -2783,49 +2783,55 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
...
-            }
-        }
-        PartitionedTopicMetadata partitionMetadata = getPartitionedTopicMetadata(topicName,
-                authoritative, false);
-        if (!topicName.isPartitioned() && partitionMetadata.partitions > 0) {
-            log.warn("[{}] Not supported calculate backlog size operation on partitioned-topic {}",
-                    clientAppId(), topicName);
-            asyncResponse.resume(new RestException(Status.METHOD_NOT_ALLOWED,
-                    "calculate backlog size is not allowed for partitioned-topic"));
+            future = validateGlobalNamespaceOwnershipAsync(namespaceName);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2788]: We need to handle validate global namespace ownership exception.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 5]
```diff
@@ -2783,49 +2783,55 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
...
+                                return;
+                            }
+                            try {
+                                ManagedLedgerImpl managedLedger = (ManagedLedgerImpl) topic.getManagedLedger();
+                                if (messageId.getLedgerId() == -1) {
+                                    asyncResponse.resume(managedLedger.getTotalSize());
+                                } else {
+                                    asyncResponse.resume(managedLedger.getEstimatedBacklogSize(pos));
+                                }
+                            } catch (WebApplicationException wae) {
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2816]: Why we catch here ?

**Review by mattisonchao — COMMENTED** [review]
I left some comment.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 2 above)
Comment (liudezhi2098) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 1 above)
Comment (liudezhi2098) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 3 above)
Comment (liudezhi2098) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 5 above)
Comment (liudezhi2098) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 6]
```diff
@@ -2783,49 +2783,56 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
...
+                                            return;
+                                        }
+                                        ManagedLedgerImpl managedLedger =
+                                                (ManagedLedgerImpl) topic.getManagedLedger();
+                                        if (messageId.getLedgerId() == -1) {
+                                            asyncResponse.resume(managedLedger.getTotalSize());
+                                        } else {
+                                            asyncResponse.resume(managedLedger.getEstimatedBacklogSize(pos));
+                                        }
+                                    });
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2821]: Missed exception handling?

Comment (liudezhi2098) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2821]: Can be captured by the upper layer，best dealt with at this layer.

**Review by codelipenghui — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]

**Pull request merged**